File: framework/cei_golang/src/impl/jsonchecker.go
package impl

import "strings"

type JsonChecker struct {
	result int
}

func NewJsonChecker() *JsonChecker {
	inst := new(JsonChecker)
	return inst
}

func (inst *JsonChecker) pass() {
	if inst.result == -1 {
		inst.result = 1
	}
}

func (inst *JsonChecker) fail() {
	inst.result = 0
}

func (inst *JsonChecker) CheckEqual(key, value string, jsonWrapper *JsonWrapper) {
	if jsonWrapper.Contains(key) {
		jsonValue := jsonWrapper.GetString(key)
		if jsonValue == value {
			inst.pass()
		} else {
			inst.fail()
		}
	}
}

func (inst *JsonChecker) CheckNotEqual(key, value string, jsonWrapper *JsonWrapper) {
	if jsonWrapper.Contains(key) {
		jsonValue := jsonWrapper.GetString(key)
		if jsonValue != value {
			inst.pass()
		} else {
			inst.fail()
		}
	}
}

func (inst *JsonChecker) ContainKey(key string, jsonWrapper *JsonWrapper) {
	if jsonWrapper.Contains(key) {
		inst.pass()
	} else {
		inst.fail()
	}
}

func (inst *JsonChecker) ValueInclude(key, value string, jsonWrapper *JsonWrapper) {
	if jsonWrapper.Contains(key) {
		tmp := jsonWrapper.GetString(key)
		if strings.Index(tmp, value) != -1 {
			inst.pass()
		}
	}
	inst.fail()
}

func (inst *JsonChecker) Complete() bool {
	return inst.result == 1
}

File: framework/cei_java/src/main/java/cei/impl/JsonChecker.java
package cei.impl;

public class JsonChecker {
    private int result = -1;

    public JsonChecker() {
    }

    private void pass() {
        if (result == -1) {
            result = 1;
        }
    }

    private void fail() {
        result = 0;
    }

    public void checkEqual(String key, String value, JsonWrapper jsonWrapper) {
        if (jsonWrapper.contains(key)) {
            String jsonValue = jsonWrapper.getString(key);
            if (jsonValue.equals(value)) {
                pass();
            }
        } else {
            fail();
        }
    }

    public void checkNotEqual(String key, String value, JsonWrapper jsonWrapper) {
        if (jsonWrapper.contains(key)) {
            String jsonValue = jsonWrapper.getString(key);
            if (!jsonValue.equals(value)) {
                pass();
            }
        }
    }

    public void containKey(String key, JsonWrapper jsonWrapper) {
        if (jsonWrapper.contains(key)) {
            pass();
        } else {
            fail();
        }
    }

    public void valueInclude(String key, String value, JsonWrapper jsonWrapper) {
        if (jsonWrapper.contains(key)) {
            String tmp = jsonWrapper.getString(key);
            if (tmp.indexOf(value) != -1) {
                pass();
                return;
            }
        }
        fail();
    }

    public boolean complete() {
        return result == 1;
    }

    public void reportError() {

    }
}
